aopy/precondition.py

```diff
@@ -1,2 +1,225 @@
 # precondition.py
-# code for preconditioning neural data
\ No newline at end of file
+# code for preconditioning neural data
+from scipy import signal
+from scipy.signal import butter, lfilter, filtfilt, windows, freqz
+import math
+import numpy as np
+import os
+import matplotlib.pyplot as plt
+#
+import nitime.algorithms as tsa
+import nitime.utils as utils
+from nitime.viz import winspect
+from nitime.viz import plot_spectral_estimate
+'''
+Filter functions
+'''
+def generate_test_signal(T, fs, freq, a=0.02):
+    '''
+
+    Args:
+        T (float): Time period in seconds
+        fs (int): Sampling frequency in Hz
+        freq (1D array): Frequencies to be mixed in the test signal. main frequency in the first element
+        plot_true (bool): boolean value if true, plots the test signal in the provided axes
+        ax (plot axis handle): axis to plot the noisy test signal
+
+    Returns:
+        x (1D array): cosine wave with multiple frequencies and noise
+    '''
+    nsamples = int(T * fs)
+    t = np.linspace(0, T, nsamples, endpoint=False)
+    # a = 0.02
+    f0 = freq[0]
+    # noise_power = 0.001 * fs / 2
+
+    x = a * 2 * np.cos(2 * np.pi * f0 * t)  # adding primary frequency
+    x += a * 0.5 * np.cos(2 * np.pi * freq[1] * t)
+    x += a * 1.5 * np.cos(2 * np.pi * freq[2] * t)
+    x += a * 20 * np.sin(2 * np.pi * 1.2 * np.sqrt(t))  # noise
+    # x += np.random.normal(scale=np.sqrt(noise_power), size=t.shape)  # noise
+
```

Comment:
Can this function only add 4 frequencies? It might be better to just make this a for loop for each input frequency with RNG terms as the amplitude.